doc/api/diagnostics_channel.md

```diff
@@ -0,0 +1,189 @@
+# Diagnostics Channel
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/diagnostics_channel.js -->
+
+The `diagnostics_channel` module provides an API to report diagnostic data
```

Comment:
Please add some description regarding the publisher and subscriber roles, which parts of the API is intended for which role and what a publisher should add in it's documentation regarding the data published.
Maybe also some notes that subscriber gets called sync so it should avoid "to lot" data processing within the callback.